pkg/netceptor/interfaces.go

```diff
@@ -0,0 +1,280 @@
+package netceptor
+
+import (
+	"context"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/ansible/receptor/pkg/logger"
+	"github.com/ansible/receptor/pkg/utils"
+	"github.com/quic-go/quic-go"
+)
+
+// -- Netceptor Interface Segregation (Dependency Injection) -----------------------------
+
+// These interfaces follow the Interface Segregation Principle by providing only
+// the methods that specific components actually need from the main Netceptor instance,
+// making the code more testable and reducing coupling.
+
+// NetcForPacketConn defines the subset of Netceptor methods needed by PacketConn.
+//
+// Implemented by:
+//   - Netceptor (main Netceptor struct implements this through its methods)
+//
+// Mock:
```

Comment:
I like most of the comments, up until Mock, I am thinking from Mock onwards this becomes a pain to maintain with diminishing returns, 
Most ide's we use can go to references, definitions etc 😄

Reply:
I agree that most of the comments are great. A couple of things:

1. I agree with Aaron that the mock references can be removed.
2. Some of the comments list specific line numbers. Those may get outdated over time as the code changes. I'd be for removing the line numbers to prevent that.

The "Used by" and "Tests" sections are super useful in helping to navigate a code base of this size.

Great job and thank you!